spec/index.html

```diff
@@ -9469,10 +9457,25 @@ <h4>Aggregate Algebra</h4>
             <p>where<br>
               &nbsp;&nbsp;M(Ψ) = [ ListEval(exprlist, μ) | μ in Ψ ]<br>
               &nbsp;&nbsp;F(Ψ) = func(M(Ψ), scalarvals), for non-<code>DISTINCT</code><br>
-              &nbsp;&nbsp;F(Ψ) = func(Distinct(M(Ψ)), scalarvals), for <code>DISTINCT</code></p>
+              &nbsp;&nbsp;F(Ψ) = func(Dedup(M(Ψ)), scalarvals), for <code>DISTINCT</code></p>
+            <p>with Dedup(M(Ψ)) being an order-preserving, duplicate-free version of the sequence M(Ψ); that is, Dedup(M(Ψ)) is a sequence of RDF terms that has the following four properties.</p>
+            <ol>
+              <li>Every unique element in M(Ψ) is contained in Dedup(M(Ψ)).</li>
+              <li>Every element in Dedup(M(Ψ)) is contained in M(Ψ).</li>
+              <li>Dedup(M(Ψ)) is free of duplicates. That is, the element at the |i|-th position in Dedup(M(Ψ)) is not the same term as the element at the |j|-th position in Dedup(M(Ψ)) for every two natural numbers |i| and |j| such that |i| &ne; |j|.</li>
+              <li>For any two elements <var>e<sub>1</sub></var> and <var>e<sub>2</sub></var> in Dedup(M(Ψ)), the relative order of their first occurrences in M(Ψ) is preserved in Dedup(M(Ψ)). That is, if <var>i<sub>1</sub></var>&nbsp;&lt;&nbsp;<var>i<sub>2</sub></var>, then <var>j<sub>1</sub></var>&nbsp;&lt;&nbsp;<var>j<sub>2</sub></var>, where
+                <ul>
+                  <li><var>i<sub>1</sub></var> is the smallest natural number such that <var>e<sub>1</sub></var> is at the <var>i<sub>1</sub></var>-th position in M(Ψ),</li>
+                  <li><var>i<sub>2</sub></var> is the smallest natural number such that <var>e<sub>2</sub></var> is at the <var>i<sub>2</sub></var>-th position in M(Ψ),</li>
+                  <li><var>j<sub>1</sub></var> is the position of <var>e<sub>1</sub></var> in Dedup(M(Ψ)), and</li>
+                  <li><var>j<sub>2</sub></var> is the position of <var>e<sub>2</sub></var> in Dedup(M(Ψ)).</li>
```

Comment:
```suggestion
              &nbsp;&nbsp;<code>F(<var>Ψ</var>) = func(Dedup(M(<var>Ψ</var>)), scalarvals)<code>, for <code>DISTINCT</code></p>
            <p>with <code>Dedup(M(<var>Ψ</var>))<code> being an order-preserving, duplicate-free version of the sequence <code>M(<var>Ψ</var>)</code>; that is, <code>Dedup(M(<var>Ψ</var>))</code> is a sequence of RDF terms that has the following four properties.</p>
            <ol>
              <li>Every unique element in <code>M(<var>Ψ</var>)</code> is contained in <code>Dedup(M(<var>Ψ</var>))<code>.</li>
              <li>Every element in <code>Dedup(M(<var>Ψ</var>))</code> is contained in <code>M(<var>Ψ</var>)</code>.</li>
              <li><code>Dedup(M(<var>Ψ</var>))</code> is free of duplicates. That is, the element at the |i|-th position in <code>Dedup(M(<var>Ψ</var>))</code> is not the same term as the element at the |j|-th position in <code>Dedup(M(<var>Ψ</var>))</code> for every two natural numbers |i| and |j| such that |i| &ne; |j|.</li>
              <li>For any two elements <var>e<sub>1</sub></var> and <var>e<sub>2</sub></var> in <code>Dedup(M(<var>Ψ</var>))</code>, the relative order of their first occurrences in <code>M(<var>Ψ</var>)</code> is preserved in <code>Dedup(M(<var>Ψ</var>))</code>. That is, if <var>i<sub>1</sub></var>&nbsp;&lt;&nbsp;<var>i<sub>2</sub></var>, then <var>j<sub>1</sub></var>&nbsp;&lt;&nbsp;<var>j<sub>2</sub></var>, where
                <ul>
                  <li><var>i<sub>1</sub></var> is the smallest natural number such that <var>e<sub>1</sub></var> is at the <var>i<sub>1</sub></var>-th position in <code>M(<var>Ψ</var>)</code>,</li>
                  <li><var>i<sub>2</sub></var> is the smallest natural number such that <var>e<sub>2</sub></var> is at the <var>i<sub>2</sub></var>-th position in <code>M(<var>Ψ</var>)</code>,</li>
                  <li><var>j<sub>1</sub></var> is the position of <var>e<sub>1</sub></var> in <code>Dedup(M(<var>Ψ</var>))</code>, and</li>
                  <li><var>j<sub>2</sub></var> is the position of <var>e<sub>2</sub></var> in <code>Dedup(M(<var>Ψ</var>))</code>.</li>
```

Reply:
@TallTed I agree with your suggestions to markup variables in the formalisms by using `<var>` tags (see also #102). I didn't do it in this PR because I wanted to avoid overloading the PR with too many types of changes at the same time.

However, I don't agree with the other type of change in your suggestions here, namely to markup formulas and mathematical functions (such as Dedup, F, and M) by using `<code>` tags. I mean, I'm in favor of using `<code>` tags for language keywords and for AST symbols, but not also for the mathematical symbols and formulas. These are not code that should appear as typewriter/monospaced font.

Reply:
I think we should strive for the best presentation we can achieve.

Is there another tag that should be used? Or another class to apply to such code tags, similar to what @afs mentioned for sparql code highlighting?

Formulas and mathematical functions that are written up in textbooks on these and related subjects are not generally printed using body text (nor, admittedly, monospaced/typewriter) styles. I have hopes we can come close with what we produce.

Reply:
@TallTed I am with you. Formulas and snippets thereof (e.g., names of mathematical functions) should be typeset to look different than the regular text. I have created issue #103 as a reminder to implement such a change throughout the whole document.

Now, are you okay with leaving such mark up out of this PR?  I would like to avoid overloading the PR with too many types of changes, in particular because I am worried that we will end up discussing the particular markup to be used, which should better be done in a separate PR.

Reply:
@TallTed 
>  I confess to flagging things when and where I notice them, and trying to fix them as quickly as possible, which often means first-attempt goes into an inappropriate PR or issue, instead of creating a fresh issue straight-off. I'll continue to try to do better at raising those distinct issues. :-)

No worries! Your reviews are much appreciated.

Reply:
I am marking this conversation (included the change suggestion) as resolved. The suggested changes will be picked up again in the context of issues #101, #102, and #103.